shared/gh/js/controller/gh.config.js

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright 2014 Digital Services, University of Cambridge Licensed
+ * under the Educational Community License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['exports'], function(exports) {
+
+    // Cache the application configuration
+    var configuration = {};
```

Comment:
We should probably move this to the config API to avoid having to think twice about where to get what config values. We should also hook this into the page setup in gh.api.js, just after the `getMe` call, as we'll need configuration every time. The configuration itself can be cached in the config api file and a function like `getConfigValue` can be used to retrieve values.

I think that `getConfigValue` function can get unnecessarily complex if the config file structure is different from the config returning from the backend, so maybe a bit more investigation into that is required.